### PR TITLE
[BE] Improve test_transformers test structure

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

We have a vast majority of test that only run on cuda. Decorating with @onlycuda causes pytest to instantiate 2x the tests and skip half of them. This overhead is non trivial when the #tests cross larger like it has for this file.

This breaks up the cuda only tests into a separate class

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -1452,7 +1452,49 @@ def test_mem_efficient_fail_sm90(self, device):
...
 class TestSDPA(NNTestCase):
-    """ Used to test the functionality of scaled_dot_product_attention
+    """ Used to test generic functionality of scaled_dot_product_attention
+    Summary:
+        If you are adding a new test to this class, make sure that it runs
+        for both cpu and cuda. If you're test is only applicable to cuda,
+        add it to TestSDPACudaOnly.
+    """
+    @parametrize("contiguous_inputs", [True, False])
+    def test_sdp_math_gradcheck(self, device, contiguous_inputs: bool):
```
Comment (mikaylagawarecki) [inline test/test_transformers.py:1462]: I think this function wasn't deleted below, it's [here](https://github.com/pytorch/pytorch/blob/0de36bc9b05e1b6c869de39b59b1d14189a69a90/test/test_transformers.py#L1735)

**Review by mikaylagawarecki — APPROVED** [review]
Looks good, just one small nit :)

File: test/test_transformers.py (hunk 1 above)
Comment (drisspg) [reply test/test_transformers.py:1462]: ahh thank you!

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**